<launch>

    <arg name="image" default="/camera/color/image_raw"/>
    <arg name="person_3d_locate_disable" default="false"/>

    <!-- DETECTNET -->
	<arg name="model_name" default="ssd-mobilenet-v2"/>
	<arg name="model_path" default=""/>
	<arg name="prototxt_path" default=""/>
	<arg name="class_labels_path" default=""/>
	<arg name="input_blob" default=""/>
	<arg name="output_cvg" default=""/>
	<arg name="output_bbox" default=""/>
	<arg name="overlay_flags" default="box,labels,conf"/>
	<arg name="mean_pixel_value" default="0.0"/>
	<arg name="threshold" default="0.5"/>
    
    <node name="person_3d_locator" pkg="person_3d_locator" type="person_3d_locator_node" output="screen">
        <rosparam file="$(find person_3d_locator)/param/person_3d_locator_param.yaml" />
        <param name="color_image_topic" type="string" value="$(arg image)"/>
        <param name="model_name" value="$(arg model_name)"/>
	<param name="model_path" value="$(arg model_path)"/>
	<param name="prototxt_path" value="$(arg prototxt_path)"/>
	<param name="class_labels_path" value="$(arg class_labels_path)"/>
	<param name="input_blob" value="$(arg input_blob)"/>
	<param name="output_cvg" value="$(arg output_cvg)"/>
	<param name="output_bbox" value="$(arg output_bbox)"/>
	<param name="overlay_flags" value="$(arg overlay_flags)"/>
	<param name="mean_pixel_value" value="$(arg mean_pixel_value)"/>
	<param name="threshold" value="$(arg threshold)"/>
        <param name="person_3d_locate_disable" value="$(arg person_3d_locate_disable)" />
    </node>
    <!-- remap from="/person_3d_locator/nearest_person" to="/move_base_simple/goal"/-->
</launch>

